### ISSUE TITLE
Backprop through LSE

### ISSUE THREAD

**Issue by abf149** [Dao-AILab/flash-attention#889]
Hello,

Does the FlashAttention backwards kernel currently support backpropagation through the Log-Sum-Exponent (LSE) output?

i.e., suppose I am training an LM & I want a penalty term against large LSE values to be in my loss function. During training, I would need to backpropagate this error term through each attention layer's LSE, in order for this penalty term to actually affect the training process. Do the backward kernels currently support this?

**Comment by 311dada** [comment]
Look forward to this PR!
